plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/gpe/GpeConvertCommandHandler.java

```diff
@@ -0,0 +1,51 @@
+
+package com.google.cloud.tools.eclipse.appengine.compat.gpe;
+
+import com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet;
+import com.google.cloud.tools.eclipse.sdk.ui.preferences.CloudSdkPrompter;
+import com.google.cloud.tools.eclipse.ui.util.ProjectFromSelectionHelper;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+
+public class GpeConvertCommandHandler extends AbstractHandler {
+
+  @Override
+  public Object execute(ExecutionEvent event) throws ExecutionException {
+    Shell shell = HandlerUtil.getActiveShellChecked(event);
+    try {
+      IProject project = ProjectFromSelectionHelper.getFirstProject(event);
+      if (project == null) {
+        throw new NullPointerException("Convert menu enabled for non-project resources");
+      }
+
+      // prompt user if Cloud SDK is not configured
+      if (CloudSdkPrompter.getCloudSdkLocation(shell) == null) {
+        // no further action required: user chose to not configure
+        return null;
+      }
+
+      IFacetedProject facetedProject = ProjectFacetsManager.create(project,
+          true /* convert to faceted project if necessary */, null /* no monitor here */);
+      if (AppEngineStandardFacet.hasFacet(facetedProject)) {
+        throw new IllegalStateException("Convert menu enabled for App Engine projects");
+      }
+
+      GpeConvertJob job = new GpeConvertJob(facetedProject);
+      job.setUser(true);
+      job.schedule();
+    } catch (CoreException ex) {
+      ErrorDialog.openError(shell, "Conversion Failed", "Failed to convert to a faceted project",
```

Comment:
Is StatusManager preferred for this nowadays?

Reply:
You're right; SM also updates as further errors are accumulated.